eip-4626.md

```diff
@@ -0,0 +1,148 @@
+---
+eip: 4626
+title: Tokenized Vault Standard
+description: A standard for tokenized vaults with a single underlying ERC20 token.
+author: Joey Santoro (@joeysantoro), t11s (@transmissions11), Jet Jadeja (@JetJadeja)
+discussions-to: https://ethereum-magicians.org/t/eip-4626-yield-bearing-vault-standard/7900
+status: Draft
+type: Standards Track
+category: ERC
+created: 2021-12-22
+---
+
+## Abstract
+
+The following standard allows for the implementation of a standard API for tokenized vaults with a single underlying ERC20 within smart contracts. This standard is an extension on the ERC20 token that provides basic functionality for depositing and withdrawing tokens and reading balances.
+
+## Motivation
+
+Tokenized vaults have a lack of standardization leading to diverse implementation details. Some various examples include lending markets (Compound, Aave, Fuse), aggregators (Yearn, Rari Vaults, Idle), and intrinsically interest bearing tokens (xSushi). This makes integration difficult at the aggregator or plugin layer for protocols which need to conform to many standards. This forces each protocol to implement their own adapters which are error prone and waste development resources.
+
+A standard for tokenized vaults will allow for a similar cambrian explosion to ERC-20, unlocking access to yield and other strategies in a variety of applications with little specialized effort from developers.
+
+
+## Specification
+
+All tokenized vaults MUST implement ERC20. If a vault is to be non-transferrable, it MAY revert on calls to transfer or transferFrom. The ERC20 operations balanceOf, transfer, totalSupply, etc. operate on the vault "shares" which represent ownership in the underlying.
+
+### Methods
+
+#### deposit
+
+`function deposit(address _to, uint256 _value) public returns (uint256 _shares)`
```

Comment:
Any way that address to send shares to can be the second parameter? In yearn vaults, we have them in this order because in decreasing order of usage:
- deposit everything for yourself
- deposit a specific amount for yourself
- deposit a specific amount and give to someone else

Depositing everything to another person is rare

Reply:
Same for `withdraw`

Reply:
cool with me, makes sense in languages (vyper) that have optional args

Reply:
I have a decent preference for the other order given it follows ERC20 spec. Wouldn't yearn need to still make adapters for the remaining functions anyway? What is the gain by switching if its only limited backward compatibility

Reply:
As a side note, the likelihood any protocol makes an adapter is quite low, I didn't make the suggestion for backwards compatibility reasons.

Anyways, I don't think ERC20 plays a huge role here in this spec in terms of driving the interface of these operations. Merely consider that the addition of override functions might be a good way to reduce gas costs for common operations, if the ordering is correct.

So adding:
```solidity
function deposit() public returns (uint256 _shares) // Deposit all to self
function deposit(uint256 _value) public returns (uint256 _shares) // Deposit some to self
function deposit(uint256 _value, address _to) public returns (uint256 _shares) // Deposit some to other
```

As another side note... I really wish people would start writing ERC specifications in YAML per [EIP-2069](https://eips.ethereum.org/EIPS/eip-2069) but that never really took off :disappointed: